hoomd/hdf5.py

```diff
@@ -358,3 +359,58 @@ def _write_header(self, f):
                             hoomd.context.current.device.cpp_msg.error("Quantity position is not vacant.")
                             raise RuntimeError("Error updating quantities with log_hdf5.")
                     data_set.attrs[quantities[i]] = i
+
+
+# For the HDF5 logger, I inherit from the GSDLogWriter.
+# This makes the assumption that the GSDLogWriter creates with the log function
+# a dictionary that has the names as keys and np.arrays as values.
+# HDF5 has the restriction that the dimension of the matrices cannot change.
+# This translates to the restriction that the dimensions of the logged quantities cannot change during a simulation run.
+class HDF5LogWriter(hoomd.dump.GSDLogWriter):
+    def __init__(self,logger):
+        super().__init__(logger)
+
+    def _write_frame(self,h5file):
+
+        f = None
+        if hoomd.context.current.device.comm.rank == 0:
+        f = self.h5file
+
+        log_dict = self.log()
+        for key in log_dict.keys():
+            if f is not None:  # Only the root rank further process the received data
+                data = log_dict[key]
+                # Check the returned object
+                if not isinstance(data, numpy.ndarray):
```

Comment:
There are many scalar (float, int) quantities that are loggable currently. Do you not plan to support them with this back-end?

Reply:
My assumption was that the dictionary contains only numpy arrays as values because the GSDDumpWriter does a similar assumption. And this was meant as a fall back test to double check is assumption.
I changed it now into a varaition, where I try to force convert the value into a numpy array and later check if it is an array of numerical data to exclude things like objects and such.

This covers now also simple float or int values as 1x1 arrays.

I excluded strings for now, because I think strings probably have a tendency to change in length which is not easy to handle for HDF5.
If we want to change that later on, it is a trivial fix.